lib/engine/game/g_1822/stock_market.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require_relative '../../stock_market'
+
+module Engine
+  module Game
+    module G1822
+      class StockMarket < Engine::StockMarket
+        def move_up(corporation)
+          r, c = corporation.share_price.coordinates
+
+          if r.positive? && share_price(r - 1, c)
+            r -= 1
+          else
+            r += 1
+            c += 1
+          end
```

Comment:
Do you need to account for this condition?

> At the end of the first operating round following a stock round in which a major company’s stock market value
reaches the game end value of the stock market.